model_server_tools/model_packaging/export_model_utils.py

```diff
@@ -0,0 +1,289 @@
+        """
+        Function to find unique model params file
+        :param files:
+        :param suffix:
+        :return:
+        """
+        match = [f for f in files if f.endswith(suffix)]
+        count = len(match)
+
+        if count == 0:
+            return None
+        elif count == 1:
+            return match[0]
+        else:
+            params = {
+                '.params': ('.params', 'parameter file'),
+                '-symbol.json': ('...-symbol.json', 'symbol file'),
+                '.onnx': ('.onnx', 'ONNX model file'),
+            }[suffix]
+
+            message = "model-export-tool expects only one %s file. Please supply the single %s file you wish to " \
+                      "export." % (params[0], params[1])
+
+            logging.error(message)
+            sys.exit(1)
+
+    @staticmethod
+    def convert_onnx_model(model_path, onnx_file):
+        """
+        Util to convert onnx model to MXNet model
+        :param model_path:
+        :param onnx_file:
+        :return:
+        """
+        try:
+            import mxnet as mx
+        except ImportError:
+            logging.error("MXNet package is not installed. Run command : pip install mxnet to install it. ")
+            sys.exit(1)
+
+        try:
+            import onnx
+        except ImportError:
+            logging.error("Onnx package is not installed. Run command : pip install mxnet to install it. ")
+            sys.exit(1)
+
+        from mxnet.contrib import onnx as onnx_mxnet
+        model_name = os.path.splitext(os.path.basename(onnx_file))[0]
+        symbol_file = '%s-symbol.json' % model_name
+        params_file = '%s-0000.params' % model_name
+        signature_file = 'signature.json'
+        # Find input symbol name and shape
+        model_proto = onnx.load(os.path.join(model_path, onnx_file))
+        graph = model_proto.graph
+        _params = set()
+        for tensor_vals in graph.initializer:
+            _params.add(tensor_vals.name)
+
+        input_data = []
+        for graph_input in graph.input:
+            shape = []
+            if graph_input.name not in _params:
+                for val in graph_input.type.tensor_type.shape.dim:
+                    shape.append(val.dim_value)
+                input_data.append((graph_input.name, tuple(shape)))
+
+        sym, arg_params, aux_params = onnx_mxnet.import_model(os.path.join(model_path, onnx_file))
+        # UNION of argument and auxillary parameters
+        params = dict(arg_params, **aux_params)
+        # rewrite input data_name correctly
+        with open(os.path.join(model_path, signature_file), 'r') as f:
+            data = json.loads(f.read())
+            data['inputs'][0]['data_name'] = input_data[0][0]
+        with open(os.path.join(model_path, signature_file), 'w') as f:
+            f.write(json.dumps(data, indent=2))
+
+        with open(os.path.join(model_path, symbol_file), 'w') as f:
+            f.write(sym.tojson())
```

Comment:
Should we blindly overwrite sym file?

Reply:
This logic was there in the existing export tool as well. Did not modify this section of the code as it was dealing with the onnx conversion.